File: 3513.number-of-unique-xor-triplets-i.java
#
# @lc app=leetcode id=3513 lang=java
#
# [3513] Number of Unique XOR Triplets I
#
# @lc code=start
import java.util.*;
class Solution {
    public int uniqueXorTriplets(int[] nums) {
        int n = nums.length;
        Set<Integer> unique = new HashSet<>();
        int[] prefixXor = new int[n + 1];
        prefixXor[0] = 0;
        for (int i = 0; i < n; i++) {
            prefixXor[i + 1] = prefixXor[i] ^ nums[i];
        }
        for (int i = 0; i < n; i++) {
            for (int k = i; k < n; k++) {
                int xorVal = prefixXor[k + 1] ^ prefixXor[i];
                unique.add(xorVal);
            }
        }
        return unique.size();
    }
}
# @lc code=end

File: 3513.number-of-unique-xor-triplets-i.go
#
# @lc app=leetcode id=3513 lang=golang
#
# [3513] Number of Unique XOR Triplets I
#
# @lc code=start
func uniqueXorTriplets(nums []int) int {
    uniqueXor := make(map[int]struct{})
    n := len(nums)
    
    // Iterate over all possible pairs (i, j) and calculate the XOR with each k >= j
    for i := 0; i < n; i++ {
        xorIJ := 0
        for j := i; j < n; j++ {
            xorIJ ^= nums[j]
            // For k >= j, xorIJ remains unchanged
            for k := j; k < n; k++ {
                uniqueXor[xorIJ] = struct{}{}
            }
        }
    }
    
    // The number of unique XOR values is the size of the map keys
    return len(uniqueXor)
}
# @lc code=end